src/charwise/builder.rs

```diff
@@ -358,7 +356,10 @@ impl CharwiseDoubleArrayAhoCorasickBuilder {
 
     #[inline(always)]
     fn get_prev(&self, i: u32) -> u32 {
-        self.states[i as usize].output_pos().unwrap()
+        self.states[i as usize]
+            .output_pos()
+            .map(|x| x.get())
+            .unwrap_or(0)
```

Comment:
```suggestion
        self.states[i as usize].output_pos().unwrap().get()
```

`output_pos().is_none()` is used to identify `is_fixed` and, if `output_pos().is_none()=true` here, it should panic.
(`output_pos()` never has 0 in the construction because the root index is always fixed.)